Examples/ObjectiveC/DDSLayerSelectionExample.m

```diff
@@ -0,0 +1,100 @@
+//
+//  DDSLayerSelectionExample.m
+//  Examples
+//
+//  Created by Jordan Kiley on 3/21/17.
+//  Copyright © 2017 Mapbox. All rights reserved.
+//
+
+// url to use: 
+
+#import "DDSLayerSelectionExample.h"
+@import Mapbox;
+
+NSString const *MBXExampleDDSLayerSelection = @"DDSLayerSelectionExample";
+
+@interface DDSLayerSelectionExample () <MGLMapViewDelegate, UIGestureRecognizerDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+
+@end
+
+@implementation DDSLayerSelectionExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds];
+    self.mapView.delegate = self;
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(39.232253141714885,-97.91015624999999)];
+    
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
+    [self.view addSubview:self.mapView];
+    
+    // Add a tap gesture recognizer to the map view.
+    UITapGestureRecognizer *gesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleTap:)];
+    gesture.delegate = self;
+    gesture.numberOfTapsRequired = 1;
+    [self.mapView addGestureRecognizer:gesture];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    
+    // Load a tileset containing U.S. states and their population density.
+    NSURL *url = [NSURL URLWithString:@"mapbox://examples.69ytlgls"];
+    
+    MGLVectorSource *source = [[MGLVectorSource alloc] initWithIdentifier:@"state-source" configurationURL:url];
+    [style addSource:source];
+    
+    MGLFillStyleLayer *layer = [[MGLFillStyleLayer alloc] initWithIdentifier:@"state-layer" source:source];
+    
+    // Access the tileset layer.
+    layer.sourceLayerIdentifier = @"stateData_2-dx853g";
+    
+    // Create a stops dictionary. This defines the relationship between population density and a UIColor.
+    NSDictionary *stops = @{
+                            @0: [MGLStyleValue valueWithRawValue:[UIColor yellowColor]],
+                            @600: [MGLStyleValue valueWithRawValue:[UIColor redColor]],
+                            @1200: [MGLStyleValue valueWithRawValue:[UIColor blueColor]]
+                            };
+    
+    // Style the fill color using the stops dictionary, exponential interpolation mode, and the feature attribute name.
+    layer.fillColor = [MGLStyleValue valueWithInterpolationMode:MGLInterpolationModeExponential sourceStops:stops attributeName:@"density" options:@{MGLStyleFunctionOptionDefaultValue : [MGLStyleValue valueWithRawValue:[UIColor whiteColor]]}];
+    
+    // Insert the new layer below the layer that contains state border lines.
+    MGLStyleLayer *symbolLayer = [style layerWithIdentifier:@"admin-3-4-boundaries"];
```

Comment:
I assume this layer is specific to Mapbox styles — let’s clarify any limitations around that.